### PR TITLE
RDS - Terraform

### PR DESCRIPTION
**Pull request by JFer11**
### Description

This pull request contains Terraform code that creates a PostgreSQL RDS instance with automated backups replication across regions using KMS encryption. The RDS instance is configured with a DB option group, DB parameter group, and VPC security group. Performance Insights and CloudWatch logs are enabled, and tags are set for all resources to aid in resource management and organization.

Additionally, the code creates a KMS key for encryption and uses a separate AWS provider for the replica region. The KMS key is aliased with the name "libera-postgresql" and has a tag set to "Environment=production".

The AWS provider is configured to use the "replica" alias and the region "us-west-2". The RDS instance is provisioned with an instance class of "db.t4g.large", allocated storage of 20GB, and the ability to scale up to a maximum of 100GB.

The maintenance window is set to run on Mondays from midnight to 3:00am, and the backup window is set to run from 3:00am to 6:00am. The retention period for automated backups is set to 7 days, and the skip_final_snapshot parameter is set to true to avoid taking a final snapshot when the RDS instance is deleted.

Overall, this code aims to provide a secure, scalable, and efficient solution for managing PostgreSQL RDS instances in a production environment.

### CODE REVIEW

File: terraform/7-rds.tf [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  identifier = "libera-postgresql"
+
+  # All available versions: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts
+  engine               = "postgres"
+  engine_version       = "14"
+  family               = "postgres14" # DB parameter group
+  major_engine_version = "14"         # DB option group
+  instance_class       = "db.t4g.large"
+
+  allocated_storage     = 20
```
Comment (jstaricco10) [inline terraform/7-rds.tf:13]: Agregate un comentario para dejar claro que son gb

File: terraform/7-rds.tf [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  create_db_option_group = true
+  create_db_parameter_group = true
+
+
+  # NOTE: Do NOT use 'user' as the value for 'username' as it throws:
+  # "Error creating DB Instance: InvalidParameterValue: MasterUsername
+  # user cannot be used as it is a reserved word used by the engine"
+  db_name  = "liberadbname"
+  username = "liberadbusername"
+  port     = 5432
```
Comment (jstaricco10) [inline terraform/7-rds.tf:25]: esta bien que este harcodeado asi el puerto?

File: terraform/7-rds.tf [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  db_subnet_group_name   = module.vpc.database_subnet_group
+  vpc_security_group_ids = [module.security_group.security_group_id]
+
+  maintenance_window              = "Mon:00:00-Mon:03:00"
+  backup_window                   = "03:00-06:00"
+  enabled_cloudwatch_logs_exports = ["postgresql", "upgrade"]
+  create_cloudwatch_log_group     = true
+
+  backup_retention_period = 7
+  skip_final_snapshot     = true  # TODO Switch
```
Comment (jstaricco10) [inline terraform/7-rds.tf]: porque ese TODO Switch mano?

File: terraform/7-rds.tf [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  deletion_protection     = false  # TODO Switch
+
+  performance_insights_enabled          = true
+  performance_insights_retention_period = 7
```
Comment (jstaricco10) [inline terraform/7-rds.tf]: en este tipo de variables pa que sea mas claro agregaria un comentario con que medida es. En este caso son dias no?

File: terraform/7-rds.tf [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  create_monitoring_role                = true
+  monitoring_interval                   = 60
```
Comment (jstaricco10) [inline terraform/7-rds.tf]: mismo

File: terraform/7-rds.tf [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    "Sensitive" = "low"
+  }
+}
+
+module "security_group" {
+  source  = "terraform-aws-modules/security-group/aws"
+  version = "~> 4.0"
+
+  name        = "libera-postgresql"
+  description = "Complete PostgreSQL example security group"
```
Comment (jstaricco10) [inline terraform/7-rds.tf]: Esto creo que te quedo asi de algun lado, le cambiaria la descripcion

**Review by jstaricco10 — COMMENTED** [review]
en general solido, aunque esta claro que sos un coco descansa. Una duda, no me queda claro la funcion de KMS... care to explain 🥥 ?

File: terraform/7-rds.tf (hunk 1 above)
Comment (JFer11) [reply terraform/7-rds.tf:13]: :scaring fabian:

File: terraform/7-rds.tf (hunk 2 above)
Comment (JFer11) [reply terraform/7-rds.tf:25]: Nope, it is not necessary

File: terraform/7-rds.tf (hunk 4 above)
Comment (JFer11) [reply terraform/7-rds.tf]: Added!

File: terraform/7-rds.tf (hunk 6 above)
Comment (JFer11) [reply terraform/7-rds.tf]: Changed!

File: terraform/7-rds.tf (hunk 3 above)
Comment (JFer11) [reply terraform/7-rds.tf]: Because during testing, it would be a mess to have that turned on, I just added a new comment explaining that.

**Comment by JFer11** [comment]
> en general solido, aunque esta claro que sos un coco descansa. Una duda, no me queda claro la funcion de KMS... care to explain 🥥 ?

Hola Coco, me alegro que preguntaras, como se que te dejaste tus IQ en Baires trash te cuento...
La key kms debe ser pasada al recurso `aws_db_instance_automated_backups_replication`, es obligatorio, si no no se puede realizar correctamente la replicación del backup entre distintas zonas.

**Review by jstaricco10 — APPROVED** [review]

**Review by jopealgorta — APPROVED** [review]

**Pull request merged**